gui/codeeditor.h

```diff
@@ -68,7 +68,7 @@ class CodeEditor : public QPlainTextEdit {
     explicit CodeEditor(QWidget *parent);
     CodeEditor(const CodeEditor &) = delete;
     CodeEditor &operator=(const CodeEditor &) = delete;
-    ~CodeEditor();
+    virtual ~CodeEditor() override;
```

Comment:
hmm... I am trying to understand the compiler warning(s).

What mistakes can you prevent with this "override" keyword? Does it actually override the base class destructor, that should be executed also right?

Cppcheck has a check that will write warnings if there is a missing "virtual" for destructors in base classes. But well as far as I know, `CodeEditor` is not a base class.

Reply:
https://stackoverflow.com/questions/34786769/defining-own-destructor-although-class-derives-from-qobject/34787463#34787463
Link provides a good explanation. It's seems best to be consistent and keep the QObject tree clean.

Reply:
I have read a little about `override`. So it ensures that the base class destructor is virtual. This sounds good, I like it!

I am not convinced about the added `virtual` keyword when there is no derived class.

Reply:
> I am not convinced about the added virtual keyword when there is no derived class.

It does not do harm. But I would like to trust Cppcheck on this.

Reply:
> https://stackoverflow.com/questions/34786769/defining-own-destructor-although-class-derives-from-qobject/34787463#34787463
> Link provides a good explanation. It's seems best to be consistent and keep the QObject tree clean.

I do not understand how this is related. Yes we heavily rely on the QObject tree and don't want to mess up that.

Reply:
CodeEditor has a Qt widget as a base clase...and they all chain back to QObject. How Qt organizes hierarchy of parent/child relationships as well as garbage management when a QObject-based instance goes out of scope. Parent is responsible for cleaning up children (only in Qt - don't try this with your parents at home).

Reply:
> CodeEditor has a Qt widget as a base clase...and they all chain back to QObject. How Qt organizes hierarchy of parent/child relationships as well as garbage management when a QObject-based instance goes out of scope.

Yes. I do not think the `virtual` keyword in the children help the parent. If the base classes have virtual destructors we are safe.

> Parent is responsible for cleaning up children (only in Qt - don't try this with your parents at home).

:-)